edx_sga/static/js/src/edx_sga.js

```diff
@@ -1,5 +1,5 @@
 /* Javascript for StaffGradedAssignmentXBlock. */
-function StaffGradedAssignmentXBlock(runtime, element) {
+async function StaffGradedAssignmentXBlock(runtime, element) {
```

Comment:
```suggestion
function StaffGradedAssignmentXBlock(runtime, element) {
```

We have two asyncs here, this async function and the scripts are async.

Are we sure we need both?

The LMS XBlock runtime (which calls this funtion) may not support promises and this might crash to unhandled promise error.

If `async` is a must, we must encapsulate the async logic inside `StaffGradedAssignmentXBlock` to ensure the promise is handled properly, instead of making the `StaffGradedAssignmentXBlock` itself async.

Reply:
Knowing your concerns now, I prefer to apply this change to remove the general async declaration. [928c587](https://github.com/nelc/edx-sga/pull/1/commits/928c587e694d8d3614d0f6804bc8ad154981a5b8)

Reply:
@OmarIthawi